compiler/rustc_codegen_llvm/src/common.rs

```diff
@@ -196,15 +203,22 @@ impl<'ll, 'tcx> ConstMethods<'tcx> for CodegenCx<'ll, 'tcx> {
             .from_key(s)
             .or_insert_with(|| {
                 let sc = self.const_bytes(s.as_bytes());
-                let sym = self.generate_local_symbol_name("str");
+                let hash = self.tcx.with_stable_hashing_context(|mut hcx| {
+                    let mut hasher = StableHasher::new();
+                    s.hash_stable(&mut hcx, &mut hasher);
+                    hasher.finish::<Hash128>()
+                });
+                let sym = format!("__rust_{RUST_VERSION}_conststr_{hash:032x}");
                 let g = self.define_global(&sym, self.val_ty(sc)).unwrap_or_else(|| {
                     bug!("symbol `{}` is already defined", sym);
                 });
                 unsafe {
                     llvm::LLVMSetInitializer(g, sc);
                     llvm::LLVMSetGlobalConstant(g, True);
-                    llvm::LLVMRustSetLinkage(g, llvm::Linkage::InternalLinkage);
+                    llvm::LLVMRustSetLinkage(g, llvm::Linkage::LinkOnceODRLinkage);
+                    llvm::SetUniqueComdat(self.llmod, g);
```

Comment:
Should this apply hidden visibility like the similar code below does?

Reply:
Good idea.